### PR TITLE
Add required dependencies

### PR DESCRIPTION
**Pull request by IvanTopolcic**
Fixes issue #223 by adding missing dependencies, and removing unnecessary dependencies.

I've also changed the imports of files to be relative, so that it can pass `depcheck`. Imports of external dependencies should use `import x from "y"` while imports from local files should use `import x from "./y"`. 

This PR also adds a CI job that uses [depcheck](https://www.npmjs.com/package/depcheck) to determine if we're using dependencies that haven't been added to the package.json file.

### CODE REVIEW

File: src/helpers/utils.ts [hunk 1]
```diff
@@ -94,6 +94,7 @@ export const getRequestBuilder = (options: RequestOptions) => {
 
     if (options.headers !== undefined) {
       options.headers.forEach((value, key) => {
+        // @ts-ignore
```
Comment (maxday) [inline src/helpers/utils.ts:97]: why do we need this?

File: tsconfig.json [hunk 2]
```diff
@@ -4,7 +4,7 @@
     // "incremental": true,                   /* Enable incremental compilation */
     "target": "es6" /* Specify ECMAScript target version: 'ES3' (default), 'ES5', 'ES2015', 'ES2016', 'ES2017', 'ES2018', 'ES2019' or 'ESNEXT'. */,
     "module": "commonjs" /* Specify module code generation: 'none', 'commonjs', 'amd', 'system', 'umd', 'es2015', or 'ESNext'. */,
-    "lib": ["es2015"] /* Specify library files to be included in the compilation. */,
+    "lib": ["es2015", "dom"] /* Specify library files to be included in the compilation. */,
```
Comment (DarcyRaynerDD) [inline tsconfig.json:7]: Why do we need to add dom libraries?

Comment (IvanTopolcic) [reply tsconfig.json:7]: I'm actually not entirely sure here -- when trying to build the project, I was running into an error with Axios: `Cannot find name 'AbortSignal'.`. Importing `dom` fixed this. Is this perhaps a versioning error or something else?

File: src/helpers/utils.ts (hunk 1 above)
Comment (IvanTopolcic) [reply src/helpers/utils.ts:97]: Typescript was complaining about an `any` type here and this seemed like the simplest solution as the code was taken from datadog-ci and I wasn't very familiar with it.

File: tsconfig.json (hunk 2 above)
Comment (IvanTopolcic) [reply tsconfig.json]: Weirdly enough it seems to be working now without it, not sure what that was, but I've gone ahead and removed it for now.

File: src/helpers/utils.ts (hunk 1 above)
Comment (DarcyRaynerDD) [reply src/helpers/utils.ts:97]: It's probably better to add an explicit type. If you're not sure what that is, you can use an any type.

**Comment by nhinsch** [comment]
Per Slack convo: let's add something to the CI pipeline to catch this kind of issue in the future.

File: src/helpers/utils.ts (hunk 1 above)
Comment (IvanTopolcic) [reply src/helpers/utils.ts]: Done!

File: scripts/depcheck.sh [hunk 3]
```diff
@@ -0,0 +1,12 @@
+#!/bin/bash
+
+DEPCHECK_OUTPUT=`depcheck`
+
+if [[ `echo $DEPCHECK_OUTPUT | grep Missing | wc -l` -gt 0 ]]; then
+    echo "Found some missing dependencies."
+    echo "$DEPCHECK_OUTPUT"
+    exit 1
+    echo "`yarn add` the missing dependencies and re-commit package.json."
```
Comment (nhinsch) [inline scripts/depcheck.sh:9]: I think this line will never print since we `exit 1` right above it

Comment (IvanTopolcic) [reply scripts/depcheck.sh:9]: ah dumb mistake, thank you!

File: scripts/depcheck.sh [hunk 4]
```diff
@@ -0,0 +1,12 @@
+#!/bin/bash
```
Comment (nhinsch) [inline scripts/depcheck.sh:1]: Why do we need this shell script? Can we just call depcheck directly?

Comment (IvanTopolcic) [reply scripts/depcheck.sh:1]: Depcheck returns a bunch of other things, such as dependencies that are imported but not used, and apparently this is not 100% accurate so I did a little bit of checking for only un-imported dependencies. For instance, it does not think we are using jest.

Comment (IvanTopolcic) [reply scripts/depcheck.sh:1]: For instance, right now it's returning this:

```
Unused devDependencies
* @types/jest
```

which I think is safe to ignore because we shouldn't care too much about unused dependencies as they're not going to break anything

Comment (nhinsch) [reply scripts/depcheck.sh:1]: Are we in fact not using that devDependency? Can we remove it? I think it would be cleaner to fix any issues depcheck finds rather than adding a wrapper around it. On the other hand, if depcheck is giving inaccurate results that is concerning 🤔

Comment (nhinsch) [reply scripts/depcheck.sh:1]: If there's some issue specifically around Jest, it looks like depcheck has an `--ignores` option which might work as well

Comment (IvanTopolcic) [reply scripts/depcheck.sh]: Hmm interesting, that's a good catch. I did end up removing it and it seems to be passing. I had thought that I tested this and the tests had failed but that might have been an unrelated issue. Apparently there can be false positives, though. I've fixed the issue and tested it, should be a lot cleaner now.

> The logic of a special is not perfect. There might be false alerts. If this happens, please open an issue for us.

Comment (IvanTopolcic) [reply scripts/depcheck.sh]: Okay yeah, it looks like the tests are failing in CI now but not locally after removing `@types/jest`. Not really sure what to do here.

Comment (IvanTopolcic) [reply scripts/depcheck.sh]: I'm just going to ignore `@types/jest` for now.

File: jest.config.js [hunk 5]
```diff
@@ -1,16 +0,0 @@
-// For a detailed explanation regarding each configuration property, visit:
```
Comment (nhinsch) [inline jest.config.js:1]: Why is this file deleted?

Comment (IvanTopolcic) [reply jest.config.js:1]: Jest configurations can be specified either in the package.json file, or a jest.config.js file. Since we already have settings for it in the package.json file, it was redundant to have this file so I deleted it.

**Review by nhinsch — APPROVED** [review]
Looks great! 🌟
